ecs-cli/modules/cli/compose/entity/entity_helper.go

```diff
@@ -144,18 +144,28 @@ func createRegisterTaskDefinitionRequest(taskDefinition *ecs.TaskDefinition) *ec
 
 // Info returns a formatted list of containers (running and stopped) in the current cluster
 // filtered by this project if filterLocal is set to true
-func Info(entity ProjectEntity, filterLocal bool) (project.InfoSet, error) {
-	containers, err := collectContainers(entity, filterLocal)
+func Info(entity ProjectEntity, filterLocal bool, desiredStatus string) (project.InfoSet, error) {
+	if err := validateDesiredStatus(desiredStatus); err != nil {
+		return nil, err
+	}
+	containers, err := collectContainers(entity, filterLocal, desiredStatus)
 	if err != nil {
 		return nil, err
 	}
 	return composecontainer.ConvertContainersToInfoSet(containers), nil
 }
 
+func validateDesiredStatus(desiredStatus string) error {
+	if desiredStatus != "" && desiredStatus != ecs.DesiredStatusRunning && desiredStatus != ecs.DesiredStatusStopped {
+		return fmt.Errorf("%s is not a valid value for desired status. Please use %s or %s.", desiredStatus, ecs.DesiredStatusRunning, ecs.DesiredStatusStopped)
+	}
+	return nil
+}
```

Comment:
This seems fine for our current behavior, though we do seem to technically allowed `DesiredStatusPending` even though it never does anything, which is confusing. If we knew there'd be more future desired statuses I might suggest making this more extensible but it's good for now.